frame/support/procedural/src/lib.rs

```diff
@@ -297,52 +298,91 @@ pub fn decl_storage(input: TokenStream) -> TokenStream {
 ///
 /// # Example:
 ///
-/// ```nocompile
+/// ```ignore
 /// construct_runtime!(
 ///     pub enum Runtime where
 ///         Block = Block,
 ///         NodeBlock = node::Block,
 ///         UncheckedExtrinsic = UncheckedExtrinsic
 ///     {
-///         System: system::{Pallet, Call, Event<T>, Config<T>} = 0,
-///         Test: test::{Pallet, Call} = 1,
-///         Test2: test_with_long_module::{Pallet, Event<T>},
+///         System: frame_system::{Pallet, Call, Event<T>, Config<T>} = 0,
+///         Test: path::to::test::{Pallet, Call} = 1,
 ///
 ///         // Pallets with instances
-///         Test3_Instance1: test3::<Instance1>::{Pallet, Call, Storage, Event<T, I>, Config<T, I>, Origin<T, I>},
-///         Test3_DefaultInstance: test3::{Pallet, Call, Storage, Event<T>, Config<T>, Origin<T>} = 4,
+///         Test2_Instance1: test2::<Instance1>::{Pallet, Call, Storage, Event<T, I>, Config<T, I>, Origin<T, I>},
+///         Test2_DefaultInstance: test2::{Pallet, Call, Storage, Event<T>, Config<T>, Origin<T>} = 4,
+///
+///         // Pallets declared with `pallet` attribute macro: no need to define the parts
+///         Test3_Instance1: test3::<Instance1>,
+///         Test3_DefaultInstance: test3,
+///
+///         // with `exclude_parts` keyword some part can be excluded.
+///         Test4_Instance1: test4::<Instance1> exclude_parts { Call, Origin },
```

Comment:
```suggestion
///         Test4_Instance1: test4::<Instance1>::{ !Call, !Origin },
```

Would such a syntax be possible?

Reply:
the thing is I want to be backward compatible.

In case somebody writes: `Test: test::{Config}`, does that mean the pallet has a config part with no generic and user want to set it. or does it mean that the pallet has a config part which is maybe generic or maybe not and user want to set it.
In the first case we don't need to retrieve the parts from the pallet, which is needed for pallets with decl_module.
In the second case we need to retrieve the parts from the pallet, which is a breaking change for pallets with decl_module which don't support it.

Reply:
a way to implemented could to have something like:
`Test: test $some_keyword_or_any_token_to_indicate_that_parts_should_be_retrieved_from_pallet {Pallet, !Call}`

But also I don't know I implemented this PR so that when you blacklist/whitelist a part, it must be an existing part of the pallet.
Like you can't do `use_parts { Call }` if the pallet has no call.